### PR TITLE
Intermediate outputs of pipeline

### LINKED ISSUE #61: Intermediate outputs of pipeline
### Description

We want to store intermediate outputs of the pipeline. The user might want to store more information about the datarun besides just the found anomalies. 

### Approach

We add another collection `IO` to the DB, which has the fields `datarun` and `io`. `datarun` is a reference to the datarun that created the intermediate output and `io` is a field that stores a dictionary.
A user then could use the pipeline JSON and specify what to store, following the specifications introduced to MLBlocks [here](https://github.com/HDI-Project/MLBlocks/issues/104).

Once we have that feature of MLBlocks, we can make use of the intermediate outputs and store them as a dictionary, using primitive.variable_name as key.

The anomalies are stored in the events collection just like before, therefore those should be returned normally and not as intermediate outputs.

### PR DESCRIPTION
**Pull request by AlexanderGeiger**
resolve #61 

Improve OrionExplorer to optionally store intermediate pipeline outputs, which are specified in the pipeline JSON file, to mongoDB GridFs.

### CODE REVIEW

**Comment by dyuliu** [comment]
Hi, I find `explorer.py` has been updated in accordance with the current DBSchema, but `cli.py` is still using the previous version.

File: orion/explorer.py [hunk 1]
```diff
@@ -21,6 +24,7 @@ def __init__(self, database='orion', **kwargs):
         self.database = database
         self._db = connect(database, **kwargs)
         self._software_versions = list(freeze.freeze())
+        self._fs = GridFS(Database(self._db, self.database))
```
Comment (csala) [inline orion/explorer.py]: I think that here you could skip the `Database` object usage like so:

```
self._fs = GridFS(self._db[self.database])
```

**Comment by csala** [comment]
> Hi, I find `explorer.py` has been updated in accordance with the current DBSchema, but `cli.py` is still using the previous version. Thus, some commands could be invalid due to the changes made in `explorer.py`.

@AlexanderGeiger Can you review this comment from @dyuliu?

**Comment by AlexanderGeiger** [comment]
@dyuliu Thanks for noticing, I think you are right. 
However, as it is not specifically related to this PR, I would suggest to address that separately. Would you mind opening a new issue for it?

**Comment by csala** [comment]
Ok, merging this in, and let's cover the CLI on a later issue/PR.

**Pull request merged**